test/uk/gov/hmrc/mtdsatestsupportapi/controllers/requestParsers/validators/CreateTestBusinessValidatorSpec.scala

```diff
@@ -69,172 +91,231 @@ class CreateTestBusinessValidatorSpec extends UnitSpec with JsonErrorValidators
           Seq(AccountingTypeFormatError)
       }
     }
 
     "return RULE_COMMENCEMENT_DATE_NOT_SUPPORTED error" when {
       "the commencement date is not in the past" in new Test {
-        validator.validate(CreateTestBusinessRawData(validNino, body.update("/commencementDate", Json.toJson(localDate(now))))) shouldBe
+        validator.validate(CreateTestBusinessRawData(validNino, bodySelfEmploymentValid.update("/commencementDate", Json.toJson(localDate(now))))) shouldBe
           Seq(RuleCommencementDateNotSupported)
       }
     }
 
     "return FORMAT_POSTCODE error" when {
       "the format of the postcode field is not valid" in new Test {
-        validator.validate(CreateTestBusinessRawData(validNino, body.update("/businessAddressPostcode", JsString("badValue")))) shouldBe
+        validator.validate(CreateTestBusinessRawData(validNino, bodySelfEmploymentValid.update("/businessAddressPostcode", JsString("badValue")))) shouldBe
           Seq(PostcodeFormatError)
       }
     }
 
     "return FORMAT_LATENCY_INDICATOR error with the appropriate path" when {
-      Seq(
-        "/latencyDetails/latencyIndicator1",
-        "/latencyDetails/latencyIndicator2"
-      ).foreach(testWith)
-
-      def testWith(path: String): Unit =
+      def testWith(path: String)(implicit pos: Position): Unit = {
         s"when the format of a latency indicator is not valid for $path" in new Test {
-          validator.validate(CreateTestBusinessRawData(validNino, body.update(path, JsString("badValue")))) shouldBe
+          validator.validate(CreateTestBusinessRawData(validNino, bodyUkPropertyValid.update(path, JsString("badValue")))) shouldBe
             Seq(LatencyIndicatorFormatError.withExtraPath(path))
         }
+      }
+
+      testWith("/latencyDetails/latencyIndicator1")
+      testWith("/latencyDetails/latencyIndicator2")
     }
 
     "return MISSING_POSTCODE error" when {
       "no post code is supplied when country code is GB" in new Test {
-        validator.validate(CreateTestBusinessRawData(validNino, body.removeProperty("/businessAddressPostcode"))) shouldBe
+        validator.validate(CreateTestBusinessRawData(validNino, bodySelfEmploymentValid.removeProperty("/businessAddressPostcode"))) shouldBe
           Seq(MissingPostcodeError)
       }
     }
 
-    "return no errors" when {
-      "no post code is supplied when country code is not GB" in new Test {
+    "return FORMAT_COUNTRY_CODE error" when {
+      "a the country code is not valid" in new Test {
+        validator.validate(CreateTestBusinessRawData(validNino, bodySelfEmploymentValid.update("/businessAddressCountryCode", JsString("badValue")))) shouldBe
+          Seq(CountryCodeFormatError)
+      }
+    }
+
+    "return RULE_UNEXPECTED_BUSINESS_ADDRESS error" when {
+      def testWithForbiddenAddressField(path: String, value: String)(implicit pos: Position): Unit = {
+        s"""the forbidden $path is provided""" in new Test {
```

Comment:
We don't need the triple quotes here (+ quite a few others tests use them which don't need too also).

Reply:
Yeah I don't know what happened here. 😅

Reply:
Do you want me to go through all our other triple quoted test names?

Reply:
Yes, just delete what you can find but don't sweat about finding them all if it's a pain.